eng/referenceFromRuntime.targets

```diff
@@ -106,7 +106,7 @@
       <ReferencePath Include="@(_filteredReferencePathFromRuntimeByFileName->'%(ReferencePath)')" Condition="'%(_filteredReferencePathFromRuntimeByFileName.Aliases)' != ''" Aliases="" />
     </ItemGroup>
 
-    <Error Condition="'@(_missingReferenceFromRuntime)' != ''"
+    <Error Condition="'@(_missingReferenceFromRuntime)' != '' and '$(DesignTimeBuild)' != 'true'"
```

Comment:
@ericstj @Anipik @safern @joperezr - Any thoughts on this change? I needed to suppress this error during design-time builds, or else VS was failing to build my `src` project. If you want to repo, pull this branch and revert this change to try it yourself.

Potentially the check for `DesignTimeBuild` could be moved to skip the whole `Target` during design-time builds.

Reply:
mm I don't think this is right. We definitely have a lot of targets that don't require building for design time builds, but in general you want all targets that affect Reference and ReferencePath to not be excluded, and this target is actually affecting those. In this case, what you are removing is the error telling you that this target failed to execute in some way, so I think it is probably better to figure out why is it that you were getting this error and fix that instead. I can help take a look at a binlog of a designtime build if you have one available.

Reply:
Definitely not the end of the world to add this, but I think that by doing so you will start getting unexpected build behavior and intellisense behavior on VS, so it is better to find the root cause for this.

Reply:
I would like to understand in what scenario you were failing because of this. I agree with @joperezr statements above that we should try finding the root cause for this and fix it instead of disabling it.

Reply:
I logged https://github.com/dotnet/runtime/issues/33427 for this issue and am reverting this change in this PR.

Feel free to pull this branch and try it. I can also share a binlog of the design-time build, but it is confusing me.